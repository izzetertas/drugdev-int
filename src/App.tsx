import React, { Component } from "react";
import ApolloClient from 'apollo-boost'
import { ApolloProvider } from 'react-apollo'

import { GRAPHQL_API_URL } from './constants'

import ContactList from './components/ContactList'

import logo from "./logo.svg";

import "./App.css";

const client = new ApolloClient({ uri: GRAPHQL_API_URL })

class App extends Component {

  render() {
    return (
      <ApolloProvider client={client}>
        <ContactList />
      </ApolloProvider>
    )
  }
}

export default App;
