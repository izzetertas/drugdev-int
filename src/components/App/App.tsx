import React, { Component } from "react";
import { Switch, Route } from 'react-router-dom'

import AddContact from "../AddContact";
import ContactList from "../ContactList";
import EditContact from "../EditContact";

import "./App.css";

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Contact List App...</h1>
          <Switch>
            <Route exact path="/" component={ContactList} />
            <Route exact path="/contacts" component={ContactList} />
            <Route exact path="/contacts/new" component={AddContact} />
            <Route exact path="/contacts/:id/edit" component={EditContact} />
          </Switch>
        </header>
      </div>
    )
  }
}

export default App;
