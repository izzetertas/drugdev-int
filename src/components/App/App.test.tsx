import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { ApolloProvider } from 'react-apollo';
import ApolloClient from 'apollo-boost';

import App from './App';

it('renders without crashing', () => {
  const wrapper = renderer.create(
    <BrowserRouter>
      <ApolloProvider client={new ApolloClient({})}>
        <App />
      </ApolloProvider>
    </BrowserRouter>
  ).toJSON();
  expect(wrapper).toMatchSnapshot();
});
