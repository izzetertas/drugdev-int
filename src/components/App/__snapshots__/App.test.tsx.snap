// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Contact List App...
    </h1>
  </header>
  <div>
    loading
  </div>
</div>
`;
